src/main/java/com/dougnoel/sentinel/webdrivers/WebDriverFactory.java

```diff
@@ -92,32 +94,57 @@ public static WebDriver instantiateWebDriver() {
         String downloadDirectory = ConfigurationManager.getOptionalProperty("download");
         if (downloadDirectory != null)
             DownloadManager.setDownloadDirectory(downloadDirectory);
+        
+        String runOnGrid = ConfigurationManager.getOptionalProperty("runOnGrid");
+    	String browser = ConfigurationManager.getBrowserName();
+    	
+        if (runOnGrid != null && runOnGrid.equals("true")) {
+        	String gridUrl = ConfigurationManager.getOptionalProperty("gridUrl");
+        	String browserVersion = ConfigurationManager.getOptionalProperty("browserVersion");
+        	
+        	//use full browser version from grid in yml or command line e.g: 88.0.4324.96
+        	driver = instantiateRemoteWebDriver(browser, browserVersion, gridUrl);
+        }else {
```

Comment:
Instead of an if/else here, follow the pattern above for Saucelabs and create a GridDriverFactory class that returns a driver object and end execution by returning that driver.

Reply:
ok, will do